config/application.rb

```diff
@@ -256,6 +256,16 @@ def cache_store_config
     config.three_scale.cors.enabled = false
     config.three_scale.cors.merge!(try_config_for(:cors) || {})
 
+    config.three_scale.content_security_policy = ActiveSupport::OrderedOptions.new
+    config.three_scale.content_security_policy.merge!(try_config_for(:content_security_policy) || {})
+
+    # Convert nested portal configurations to OrderedOptions
+    [:admin_portal, :developer_portal].each do |portal|
+      portal_config = ActiveSupport::OrderedOptions.new
+      portal_config.merge!(config.three_scale.content_security_policy.send(portal))
```

Comment:
```suggestion
      portal_config.merge!(config.three_scale.content_security_policy.send(portal) || {})
```

or add some other safety mechanism, so that the app doesn't crash if the CSP yaml is not there (e.g. in dev environment, if we don't re-copy the config example folder)